antora/supplemental-ui/helpers/parseHugoMenu.js

```diff
@@ -0,0 +1,51 @@
+const fs = require("fs");
+const toml = require("toml");
+
+module.exports = () => {
+  try {
+    const filePath = "./hugo.toml";
+    const fileContent = fs.readFileSync(filePath, "utf8");
+    const config = toml.parse(fileContent);
+    const menu = config.menu.main || [];
+    const processed = new Set();
+
+    const createMenu = (item) => {
+      if (processed.has(item)) {
+        return {};
+      }
+      processed.add(item);
+      const children = menu
+        .filter((child) => child.parent === item.name)
+        .map(createMenu);
+      if (children.length) {
+        return {
+          name: item.name,
+          url: item.url,
+          pageRef: item.pageRef,
+          weight: item.weight || 0,
+          parent: item.parent,
+          graphic: item.graphic,
+          children,
+        };
+      } else {
+        return {
+          name: item.name,
+          url: item.url,
+          pageRef: item.pageRef,
+          weight: item.weight || 0,
+          parent: item.parent,
+          graphic: item.graphic,
+        };
+      }
+    };
+
+    const menuData = menu
+      .filter((item) => typeof item.parent === "undefined")
+      .map((item) => createMenu(item));
+    const sortedMenuData = menuData.sort((a, b) => a.weight - b.weight);
+    return sortedMenuData;
+  } catch (error) {
+    console.error("Error parsing TOML:", error);
```

Comment:
Might be better not to catch the error, i.e. if any error occurs it breaks the build